### DISCUSSION TITLE
Pocketbase SDK Setting Cookies in Next.js 13 Server Component

### DISCUSSION THREAD

**Discussion by Dave-lab12** [pocketbase/pocketbase#2642] [Q&A] [answered]
The problem is to set cookies correctly when making requests on the server component of Next.js 13. 
Additionally, I've encountered difficulties in manually setting the cookie due to the mismatch between the cookie formats used in Next.js 13 and the format expected by document.cookies.

## Possible workaround

<details>
    <h1 style="display:inline-block"><summary>pocketbaseServer.ts</summary></h1>
    
 ```typescript
  import PocketBase from "pocketbase"
  import { cookies } from "next/headers"
  export const pocketbase = () => {
     const cookie = cookies();
     const pocketbase = new PocketBase(process.env.NEXT_PUBLIC_POCKETBASE_URL);
     const stringifiedCookies = convertCookieString(cookie.getAll());
     pocketbase.authStore.loadFromCookie(stringifiedCookies);
     // add on change
  return pocketbase
}
  ```
  <h1>convertCookieString.ts</h1>

 ```typescript
   function convertCookieString(cookieString: any) {
    const cookies = cookieString; 

    const convertedCookies = [];

    for (const cookie of cookies) {
      const name = cookie.name;
      const value = cookie.value;

     // Encode the name and value using encodeURIComponent
     const encodedName = encodeURIComponent(name);
     const encodedValue = encodeURIComponent(value);

     const convertedCookie = `${encodedName}=${encodedValue}`;

     convertedCookies.push(convertedCookie);
    }
    const result = convertedCookies.join("; ");

   return result;
   }

```
</details>

**Comment by ganigeorgiev** [comment, marked as answer]
If you haven't already, for a general workflow you can check https://github.com/pocketbase/js-sdk#ssr-integration.
It is a little outdated but you can also check some of the suggestions from https://github.com/pocketbase/js-sdk/issues/69.
Official Next.js 13 SSR example will be added after the planned JS SDK refactoring.

**Comment by Dave-lab12** [comment]
[Found Cleaner implementation](https://github.com/pocketbase/js-sdk/issues/69#issuecomment-1312614581)  thank you 😀
